### PR TITLE
Add thanhnien scraper tutorial

### PR DESCRIPTION
**Pull request by NBQNQT**

### CODE REVIEW

File: man/thanhnien.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
+Scraping vaccine images from tuoitre
+====================================
+
+Site analysis
+-------------
+
+We scrape image and caption about vaccine from searched page from thanhnien.vn :
```
Comment (McSinyx) [inline man/thanhnien.rst]:
```suggestion
We scrape image and caption about vaccine from searched page from thanhnien.vn:

```

File: man/thanhnien.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+We scrape image and caption about vaccine from searched page from thanhnien.vn :
+
+.. code-block:: html
+
+   https://thanhnien.vn/vaccine
+
+
+For the images and their captions in the articles, we will try to get the ``src`` ---image source and 
+``alt`` ---image caption of the ``<img>`` tag.
```
Comment (McSinyx) [inline man/thanhnien.rst]:
```suggestion
For the images and their captions in the articles, we will try to get the ``src``---image source and 
``alt``---image caption of the ``<img>`` tag. 
```

File: man/thanhnien.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+For the images and their captions in the articles, we will try to get the ``src`` ---image source and 
+``alt`` ---image caption of the ``<img>`` tag. 
+
+Scraping explanation
+--------------------
+
+The site taken is site for searched for vaccine in thanhnien.vn
+.. code-block:: python
```
Comment (McSinyx) [inline man/thanhnien.rst:21]:
```suggestion

.. code-block:: python
```

File: man/thanhnien.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+   INDEX = 'https://thannien.vn/vaccine'
+	
+
```
Comment (McSinyx) [inline man/thanhnien.rst]:
```suggestion

```

File: man/thanhnien.rst [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+.. code-block:: python
+	
+   def articles(links):
+    """Find URLs contains 'vacxin' in the given link."""
+    for a in links:
+        href = a.get('href')
+        if href is None: continue
+        url = 'http://thanhnien.vn/' + href
+        if url.endswith('.html') and 'vac' in url: yield url
```
Comment (McSinyx) [inline man/thanhnien.rst]: Pls check the indentation.

File: man/thanhnien.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+		
+From searched articles we focus only on url ended with htm and vac in the caption and image
+
+
```
Comment (McSinyx) [inline man/thanhnien.rst:42]:
```suggestion
```

File: man/thanhnien.rst [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+   async def scrape_images(url, dest, client, nursery):
+    """Search for img in the articles in order to download the images."""
+    article = await client.get(url)
+    for img in parse_html5(article.text).iterfind('.//img'):
+        caption, url = img.get('alt'), img.get('data-src')
+        if caption is None or ('vắc') not in caption.lower():
+            if caption is None or ('vac') not in caption.lower(): continue
+        if url is None: url = img.get('src')
+        if url.endswith('logo.svg'): continue
+        nursery.start_soon(download, caption, url, dest, client)
```
Comment (McSinyx) [inline man/thanhnien.rst]: Same here and below.

**Review by McSinyx — CHANGES_REQUESTED** [review]

File: man/thanhnien.rst [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+       for img in parse_html5(article.text).iterfind('.//img'):
+           caption, url = img.get('alt'), img.get('data-src')
+           if caption is None or ('vắc') not in caption.lower():
+               if caption is None or ('vac') not in caption.lower(): continue
+           if url is None: url = img.get('src')
+           if url.endswith('logo.svg'): continue
+           nursery.start_soon(download, caption, url, dest, client)
+				
+The website used in vietnamese languages so we use both key words as 'vac' and 'vắc' for either vaccine
+and vắc xin.
```
Comment (McSinyx) [inline man/thanhnien.rst]:
```suggestion
The website used in vietnamese languages so we use both key words as *vac* and *vắc* for either *vaccine*
and *vắc xin*.

```

File: man/thanhnien.rst [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+       fi = await client.get(url)
+       except ConnectTimeout:
+           return
+   async with await open_file(directory/f'image{ext}', 'wb') as fo:
+       async for chunk in fi.aiter_bytes(): await fo.write(chunk)
+       await (directory/'caption').write_text(caption, encoding='utf-8')
+       print(caption)
+	
+The download function will download the image from ``src`` and the caption from ``alt``.
+Each image and its caption is  list on in folder and named "image", "caption" respectively inside a folder named by
```
Comment (McSinyx) [inline man/thanhnien.rst]:
```suggestion
Each image and its caption is list on in folder and named ``image``, ``caption`` respectively inside a folder named by
```

**Review by McSinyx — DISMISSED** [review]

**Review by pdhlong — APPROVED** [review]

**Pull request merged**